server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -532,44 +528,47 @@ private String createDN(Consumer consumer, Owner owner) {
         throws IOException {
         List<org.candlepin.model.dto.Product> products = new ArrayList<>();
         products.add(container);
-        Set<X509ByteExtensionWrapper> result = v3extensionUtil.getByteExtensions(null, products, null,  null);
-        return result;
+        return v3extensionUtil.getByteExtensions(null, products, null,  null);
     }
 
-    private byte[] createContentAccessDataPayload(Owner owner, Environment environment) throws IOException {
+    private byte[] createContentAccessDataPayload(Owner owner, Environment environment, Consumer consumer)
+        throws IOException {
         // fake a product dto as a container for the org content
-        Set<Product> containerSet = new HashSet<>();
-        Set<String> entitledProductIds = new HashSet<>();
-        List<org.candlepin.model.dto.Product> productModels = new ArrayList<>();
-        Map<String, EnvironmentContent> promotedContent = getPromotedContent(environment);
-        String contentPrefix = this.getContentPrefix(owner, environment);
         Product container = new Product();
-        Entitlement emptyEnt = new Entitlement();
-        Pool emptyPool = new Pool();
-        Product skuProduct = new Product();
-        Consumer emptyConsumer = new Consumer();
-        emptyConsumer.setOwner(owner);
-
-        containerSet.add(container);
         container.setId("content_access");
         container.setName(" Content Access");
 
         this.ownerContentCurator.getActiveContentByOwner(owner.getId())
-            .forEach((content, enabled) -> container.addContent(content, enabled));
+            .forEach(container::addContent);
 
+        Consumer emptyConsumer = new Consumer();
+        emptyConsumer.setOwner(owner);
         emptyConsumer.setEnvironment(environment);
-        emptyEnt.setPool(emptyPool);
-        emptyEnt.setConsumer(emptyConsumer);
+        // TODO Do we need any other data for content filtering?
```

Comment:
I am not aware of anything else we filter based on the consumer. @Ceiu do you?

Reply:
The only thing I know of that would be fetched from the consumer object aside from arch would be: CPU sockets/cores, ram, storage bands, etc.; nothing that applies for filtering in SCA mode, I think.

Reply:
That's all included in the facts, so we should be good